app/src/Containers/UserForm/index.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useRef, useState } from "react";
+import { passwordValidation } from "../../Helpers/password-validation";
+import { UserFormWrapper } from "./styles";
+import qs from "qs"
+
+interface user {
+  username: string,
+  password: string,
+}
+
+const reqUserCreate = async (newUser: user) => {
+  const in_flight = await fetch("/api/user/create", {
+    method: 'POST',
+    body: JSON.stringify(newUser),
+    headers: {
+      'Content-Type': "application/json"
+    }
+  });
+  const output = await in_flight.json();
+  return output
+}
+
+const reqLogin = async (user: user) => {
+  const in_flight = await fetch('/api/user/login', {
+    method: 'POST',
+    body: JSON.stringify(user),
+    headers: {
+      'Content-type': 'application/json'
+    }
+  })
+  const output = await in_flight.json()
+  return output
+}
+
+const creationErrorHandler = (res: any) => {
+  if(res.code === 11000){
+    alert("That username is already in use.")
+    return false
+  }
+}
+
+const loginErrorHandler = (res: any) => {
+  alert(res)
+}
+
+const UserForm = (props: any) => {
+  
+  const [newAccount, setNewAccount] = useState(false)
+  const usernameInput = useRef()
+  const passwordInput = useRef()
+  const passwordCheckInput = useRef()
+  const [errorMessage,setError] = useState('')
+
+  const handleCreate = (userInfo: user, passwordCheck: string) => {
+    const checkForErrors = passwordValidation(userInfo.password ,passwordCheck)
+    if(checkForErrors.length <= 0){
+      reqUserCreate(userInfo).then(res => {
+        if(!res.token) creationErrorHandler(res)
+        else {
+          props.showForm()
+          props.setToken(qs.parse(document.cookie).token)
+        }
+        
+      })
+    }
+    else if(checkForErrors){
+      setError(`The password ${checkForErrors.join(" & ")}.`)
+    }
+  }
```

Comment:
this is 4 year old react style. it's hard to know that when reading articles tho. this is a `classical` approach meaning `class` based react the old `class Comp extends React.Component {}` bullshti